src/backend/gp_libpq_fe/fe-connect.c

```diff
@@ -436,7 +436,7 @@ PQconnectStartParams(const char **keywords,
 	{
 		conn->status = CONNECTION_BAD;
 		/* errorMessage is already set */
-		return false;
+		return conn;
```

Comment:
The original code is absolutely buggy. However please double check the logic to see whether conn instead of others like NULL should be returned.